### PR TITLE
Handle request timeouts.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Use echo middleware in addition to `http.Server` for timeouts.

## Test Plan

Manual testing with a long-running query and a short timeout. I see the query canceled and a timeout response is returned.

### CODE REVIEW

**Comment by tolikzinovyev** [comment]
What changes internally? Could you fix CI?

**Comment by tolikzinovyev** [comment]
Do you get a timeout HTTP code back now?

**Comment by winder** [comment]
I'm converting this PR back to a draft.

It looks like there are a few gotchas with the middleware, and further testing is required. I found this in the middleware source:
https://github.com/labstack/echo/blob/master/middleware/timeout.go#L12
```
// WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
// WARNING: Timeout middleware causes more problems than it solves.
// WARNING: This middleware should be first middleware as it messes with request Writer and could cause data race if
// 					it is in other position
```

**Review by gmalouf — APPROVED** [review]
Discussed live - approved.

File: api/server.go [hunk 1]
```diff
@@ -17,6 +17,15 @@ import (
 	"github.com/algorand/indexer/idb"
 )
 
+const (
+	// ReadTimeout covers the time from when the connection is accepted to when the request body is fully read
+	ReadTimeout = 5 * time.Second
+	//WriteTimeout normally covers the time from the end of the request header read to the end of the response write
+	WriteTimeout = 59 * time.Second
```
[the remark this answers is not in the document]
Comment (winder) [reply api/server.go:24]: I guess it might protect us from a misbehaving client who doesn't accept the response

**Pull request merged**